native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/internal/AgentCommandLineProvider.java

```diff
@@ -82,7 +83,7 @@ public AgentCommandLineProvider() {
             if (agentOptions.stream().map(s -> s.split("=")[0]).anyMatch(s -> s.contains("config-output-dir"))) {
                 throw new IllegalStateException("config-output-dir cannot be supplied as an agent option");
```

Comment:
The idea with #190 is now to have an enum which tells what we want to generate: trace output or config output. I'm not sure what _config merge dir_ is used for. In any case, this should probably not be mixed into this PR.

Reply:
From [docs](https://www.graalvm.org/22.0/reference-manual/native-image/Agent/):
> It can be necessary to run the target application more than once with different inputs to trigger separate execution paths for a better coverage of dynamic accesses. The agent supports this with the `config-merge-dir` option which adds the intercepted accesses to an existing set of configuration files:
> ```shell
> $JAVA_HOME/bin/java -agentlib:native-image-agent=config-merge-dir=/path/to/config-dir/ ...
>                                                               ^^^^^
> ```
> 
> If the specified target directory or configuration files in it are missing when using `config-merge-dir`, the agent creates them and prints a warning.

Basically we would need to reproduce this behavior using `native-image-configure` invocation.

I agree that this should not be mixed with this PR, but it is something to keep in mind when designing tasks related to `native-image-configure` invocation.

Reply:
I see, in combination with this race condition this starts to become bind blowing.